lib/hanami/router.rb

```diff
@@ -693,6 +693,10 @@ def env_for(env, params = {}, options = {})
     # @api private
     PREFIXED_NAME_SEPARATOR = "_"
 
+    # @since 2.3.0
+    # @api private
+    NAME_UNDERSCORE_REGEXP = /[\-+~.]/.freeze
```

Comment:
I don't think the leading `\` does anything, since it just acts as an escape, and `-` doesn't need to be escaped at the beginning of a regexp. It only acts as a regexp separator when it's between two characters. I'm also fine with keeping it though, just to be explicit

Reply:
Well spottet, @cllns !

Actually, I quote `-` inside square brackets always and on purpose: Not everybody is experienced enough with Regexp to know and if you add another acceptable character to the front of the list, the unquoted `-` silently flips from being a literal to a range operator. And if this someone is out of luck, the resulting range includes the literal `-` and therefore the tests might not even fail. This quite likely wouldn't be a huge problem here, but it could introduce a difficult to spot bug or even open a security gap elsewhere. Thus, better safe than sorry.

But feel free to insist or change it if you prefer dropping the quote.